src/sentry/runner/commands/backup.py

```diff
@@ -16,9 +16,21 @@
 def import_(src):
     "Imports data from a Sentry export."
 
-    for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
-        if obj.object._meta.app_label not in EXCLUDED_APPS:
-            obj.save()
+    try:
+        with transaction.atomic():
+            for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
+                if obj.object._meta.app_label not in EXCLUDED_APPS:
+                    obj.save()
+    # There are two integrity error exceptions we handle here.
+    # 1. Foreign key check failure
+    # 2. Duplicate key violation
+    # For both, it seems reasonable to continue importing and just display a warning message
+    except IntegrityError as e:
+        click.echo(
+            ">> Are you restoring from a clean database? If not, we highly recommend doing so.",
+            err=True,
+        )
+        raise (e)
```

Comment:
Won't re-raising the integrity error stop the import?

Reply:
I believe that's the intention now, and the commend should be updated or probably just removed.

Reply:
yep, that's the intention. editing this comment to reflect that